### PR TITLE
Usage stats: Tune collector execution startup and interval

### PR DESCRIPTION
**Pull request by papagian**
**What is this feature?**

- Introduces a random delay between 30 to 120 seconds before running the stats update
- Makes the collector interval configurable
- Prevent sending reports before collector has run at least once

**Why do we need this feature?**

Delays the execution of the collector updater by a random factor so that overhead to the database backend is reduced, if multiple instances restart at the same time.

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by DanCech — COMMENTED** [review]
LGTM, just need to make sure we also delay sending reports until the stats have been collected

File: pkg/infra/usagestats/statscollector/service.go [hunk 1]
```diff
@@ -91,14 +97,21 @@ func (s *Service) RegisterProviders(usageStatProviders []registry.ProvidesUsageS
 }
 
 func (s *Service) Run(ctx context.Context) error {
-	s.updateTotalStats(ctx)
-	updateStatsTicker := time.NewTicker(time.Minute * 30)
+	sendInterval := time.Second * time.Duration(s.cfg.MetricsTotalStatsIntervalSeconds)
+	nextSendInterval := time.Duration(rand.Intn(MAX_DELAY-MIN_DELAY)+MIN_DELAY) * time.Second
```
[the remark this answers is not in the document]
Comment (papagian) [reply pkg/infra/usagestats/statscollector/service.go:101]: my intention was to introduce a random delay at the beginning and then execute all the following ones at the standard interval (eg 30s): since the first one is delayed by a random factor all of the will shift by the same factor so they won't occur exactly at the same time

[the remark this answers is not in the document]
Comment (DanCech) [reply pkg/infra/usagestats/statscollector/service.go:101]: a fixed interval is fine as long as the starting time is random

**Review by DanCech — CHANGES_REQUESTED** [review]
I like where this is going, but we need to make sure that the initial delays work together properly

File: pkg/infra/usagestats/service/service.go [hunk 2]
```diff
@@ -82,22 +85,27 @@ func (uss *UsageStats) Run(ctx context.Context) error {
...
+				if traceID, err := uss.sendUsageStats(ctx); err != nil {
+					uss.log.Warn("Failed to send usage stats", "error", err, "traceID", traceID)
+				}
+
+				lastSent = time.Now()
+				if err := uss.kvStore.Set(ctx, "last_sent", lastSent.Format(time.RFC3339)); err != nil {
+					uss.log.Warn("Failed to update last sent time", "error", err)
+				}
+
+				for _, callback := range uss.sendReportCallbacks {
```
Comment (papagian) [inline pkg/infra/usagestats/service/service.go:98]: not 100% sure if we want to block also these if the stats are not updated but I guess we don't want to send any reports in that case

File: pkg/infra/usagestats/service/service.go [hunk 3]
```diff
@@ -82,22 +85,27 @@ func (uss *UsageStats) Run(ctx context.Context) error {
...
 		select {
 		case <-sendReportTicker.C:
-			if traceID, err := uss.sendUsageStats(ctx); err != nil {
-				uss.log.Warn("Failed to send usage stats", "error", err, "traceID", traceID)
-			}
-
-			lastSent = time.Now()
-			if err := uss.kvStore.Set(ctx, "last_sent", lastSent.Format(time.RFC3339)); err != nil {
-				uss.log.Warn("Failed to update last sent time", "error", err)
+			if uss.readyToReport {
```
Comment (DanCech) [inline pkg/infra/usagestats/service/service.go:88]: Might be cleaner to just revert the changes below and use something like this?
```suggestion
			if !uss.readyToReport {
				nextSendInterval = time.Minute
				sendReportTicker.Reset(nextSendInterval)
				continue
			}
```

**Review by DanCech — APPROVED** [review]
LGTM

**Review by zserge — APPROVED** [review]
